src/vcpkg/configure-environment.cpp

```diff
@@ -109,14 +110,27 @@ namespace vcpkg
 #endif // ^^^ !VCPKG_CE_SHA
 
         Command cmd_run(node_path);
-        cmd_run.string_arg("--harmony");
         cmd_run.string_arg(ce_path);
         cmd_run.forwarded_args(args);
         if (Debug::g_debugging)
         {
             cmd_run.string_arg("--debug");
         }
 
+        if (LockGuardPtr<Metrics>(g_metrics)->metrics_enabled())
+        {
+            cmd_run.string_arg("--z-enable-metrics");
+        }
+
+        cmd_run.string_arg("--vcpkg-root").string_arg(paths.root);
```

Comment:
While passing all these parameters as cli args is fine, I want to also note that we do already put them into a json doc inside an environment variable for the recursive vcpkg case.

As an aside, that doc should probably be extended with these new properties.

Reply:
That json blob is at the wrong level of abstraction: it's in the command line parser layer, and records what the command line parser *inputs* were in hopes that the recursive invocation will re-derive the same values. But here we need the TypeScript world to consume the *outputs*.

Doing CLI args like this is also easier when debugging since you can get everything you need to do your debugging of that stuff without also needing to each the TypeScript debugger to set up all that separate environment stuff.